CMakeLists.txt

```diff
@@ -84,8 +84,19 @@ endif()
 if(NOT TARGET FARGPARSE::fargparse)
   find_package(FARGPARSE QUIET)
 endif()
+
+
+option(USE_EXTDATA2G "Use ExtData2G" OFF)
 if(NOT TARGET YAFYAML::yafyaml)
-  find_package(YAFYAML QUIET)
+  if(USE_EXTDATA2G)
+     set (EXTDATA_TARGET "MAPL.ExtData2G" CACHE STRING "ExtData Target")
+     find_package(YAFYAML REQUIRED)
+     message (STATUS "Building with ExtData2G")
+  else()
+     set (EXTDATA_TARGET "MAPL.ExtData" CACHE STRING "ExtData Target")
+     find_package(YAFYAML QUIET)
+     message (STATUS "Building with ExtData1G")
+  endif()
```

Comment:
I don't think the conditional on TARGET YAFYAML is needed for this.  (But it was not part of this change, so I could not highlight that line.)

Reply:
fixed this up per your advice